src/zimdump.cpp

```diff
@@ -416,7 +413,14 @@ int subcmdDump(ZimDumper &app,  std::map<std::string, docopt::value> &args)
         std::string nspace = args["--ns"].asString();
         filter = [nspace](const char c){ return nspace.at(0) == c; };
     }
-    return subcmdDumpAll(app, args["--dir"].asString(), redirect, filter);
+    
+    std::string directory = args["--dir"].asString();
+
+    if (*directory.rbegin() == '/'){
```

Comment:
1. What if `directory` is an empty string? (though this is a case for ultimate foolproofness)
2.  Assuming that pre-C++11 compilers are left behind, you can use [`std::string::back()`](https://en.cppreference.com/w/cpp/string/basic_string/back) here, which is more readable.

Reply:
I agree with you @veloman-yunkan, using `back()` is much more readable. If input with an empty string is given, we can either throw a warning `--dir must be provided` or we can dump in the current directory. Which of these is better?

Reply:
It is better to reject empty value for `--dir`